portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,5 +42,35 @@ private String convertToJsonUsingGson(ArrayList storeCommentList) {
     String json = gson.toJson(storeCommentList);
     return json;
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    // Get the input from the form.
+    String fName = request.getParameter("firstname");
+    String lName = request.getParameter("lastname");
+    String subject = request.getParameter("subject");
+    String message = request.getParameter("message");
+
+    String myUserComment = "Name: " + fName + " " + lName + "\n" + "Subject: " + subject + "\n" + message;
+    storeCommentList.add(myUserComment);
+
+    // Redirect back to the Contact page.
+    response.sendRedirect("contact.html");
+    
+  }
+
+  /**
+   * @return the request parameter, or the default value if the parameter
+   *         was not specified by the client
+   */
+  private String getParameter( String name) {
+    String value = name;
```

Comment:
Since name is a String why redefine it as 'value'?

Reply:
My eyes didn't really catch that. I will optimize it by simply using the name variable. I will also change the name variable to a name that fits all four inputs from the comments.
Thank you.